src/process/shell.cr

```diff
@@ -0,0 +1,81 @@
+class Process
+  # Converts a sequence of strings to one joined string with each argument shell-escaped.
+  #
+  # This is then safe to pass to the current system's shell or as part of the command in `system()`.
+  #
+  # ```
+  # files = ["my file.txt", "another.txt"]
+  # `grep -E 'fo+' -- #{Process.shell_quote(files)}`
+  # ```
+  def self.shell_quote(args : Enumerable(String)) : String
+    {% if flag?(:win32) %}
+      shell_quote_windows(args)
+    {% else %}
+      shell_quote_posix(args)
+    {% end %}
+  end
+
+  # Shorthand for `shell_quote({arg})`.
+  def self.shell_quote(arg : String) : String
+    shell_quote({arg})
+  end
+
+  # Converts a sequence of strings to one joined string with each argument shell-quoted.
+  #
+  # This is then safe to pass to a POSIX shell.
+  #
+  # ```
+  # files = ["my file.txt", "another.txt"]
+  # Process.shell_quote_posix(files) # => "'my file.txt' another.txt"
+  # ```
+  def self.shell_quote_posix(args : Enumerable(String)) : String
+    args.join(' ') do |arg|
+      if arg.empty?
+        "''"
+      elsif arg =~ %r([^a-zA-Z0-9%+,\-./:=@_]) # not all characters are safe, needs quoting
+        "'" + arg.gsub("'", %('"'"')) + "'"    # %(foo'ba#r) becomes %('foo'"'"'ba#r')
+      else
+        arg
+      end
```

Comment:
```suggestion
      case arg
      when .empty?
        "''"
      when .matches? %r([^a-zA-Z0-9%+,\-./:=@_]) # not all characters are safe, needs quoting
        "'" + arg.gsub("'", %('"'"')) + "'"      # %(foo'ba#r) becomes %('foo'"'"'ba#r')
      else
        arg
      end
```

Reply:
why

Reply:
- same indent for all branches
- `String#matches?`

Reply:
Applied `matches?`